### PR TITLE
Sorting, searching, filtering funtionality

### PR DESCRIPTION
**Pull request by IrfanSalim**
For #3 

Added a control section which contains the sorting, searching and filter functionality.

image

image

Please check the code and test it out, share any feedback if there.

### CODE REVIEW

**Review by yodkwtf — CHANGES_REQUESTED** [review]
Please move the controls logic and JSX to a separate component and import that component in the Home.jsx

Should keep the page component free from these logic handling as much as possible.

**Comment by yodkwtf** [comment]
> Please move the controls logic and JSX to a separate component and import that component in the Home.jsx
> 
> Should keep the page component free from these logic handling as much as possible.

We can have another context for this if required. Appreciate all the good work!

**Comment by IrfanSalim** [comment]
Hey, @yodkwtf 

I'm sorry, im unable to figure out this.

I'm not well versed with context and haven't used it yet, but still I tried to make it work but its not working.

I also tried to move all logic and jsx to a different component but that also is having bugs, I'm not able to figure out all props and states between both.

Please let me know what can be done, thanks.

**Comment by yodkwtf** [comment]
No worries, I'll take a pull, test your features, and merge them over the weekend. Once it's finished I'll try to do the component separation and you can refer to that code later to get an idea.

Thanks for the contribution.

**Comment by yodkwtf** [comment]
Please do some thorough testing. The search and filter aren't working as expected. If I enter a term that doesn't match with anything it should return "No movies found". Currently, it's showing all the movies.

Good work though. It looks very clean, especially the filter modal.

**Comment by IrfanSalim** [comment]
> Please do some thorough testing. The search and filter aren't working as expected. If I enter a term that doesn't match with anything it should return "No movies found". Currently, it's showing all the movies.
> 
> Good work though. It looks very clean, especially the filter modal.

Please check now, fixed that issue.

**Comment by yodkwtf** [comment]
Great work. Please add one more change to make it responsive for mobile 

image

Add **flex-wrap** and gap on the **controls** class. I'll merge once this is done. Thanks!

File: client/src/index.css [hunk 1]
```diff
@@ -323,6 +323,108 @@ footer .icon i {
   margin: 20px auto;
 }
 
+.controls {
+  display: flex;
+  justify-content: space-around;
+  align-items: center;
+  width: 100%;
+  margin-bottom: 1rem;
```
Comment (yodkwtf) [inline client/src/index.css:332]: Add flex-wrap and gap here to make it resonsive

File: client/src/index.css [hunk 2]
```diff
@@ -323,6 +323,108 @@ footer .icon i {
...
+  background: var(--primary);
+  color: #fff;
+  border: none;
+  padding: 8px 12px;
+  border-radius: 4px;
+  cursor: pointer;
+  transition: background 0.3s ease;
+}
+
+.controls button:hover {
```
Comment (yodkwtf) [inline client/src/index.css:368]: use different shade of primary btns to keep styles consistent. It must be their in some variable, add new var if needed but keep it blue only

File: client/src/index.css [hunk 3]
```diff
@@ -323,6 +323,108 @@ footer .icon i {
...
+  border: none;
+  padding: 10px;
+  margin: 10px auto;
+  border-radius: 4px;
+  cursor: pointer;
+  transition: background 0.3s ease;
+}
+
+.modal-content button:hover {
+  background: var(--dark-green);
```
Comment (yodkwtf) [inline client/src/index.css]: use diff shade of blue as mentioned

File: client/src/index.css (hunk 1 above)
Comment (IrfanSalim) [reply client/src/index.css:332]: Done.

image

File: client/src/index.css (hunk 2 above)
Comment (IrfanSalim) [reply client/src/index.css:368]: Done

File: client/src/index.css (hunk 3 above)
Comment (IrfanSalim) [reply client/src/index.css]: Done

**Pull request merged**